File: CalendarFluent.UI/Components/SortableList.razor.js
export function init(id, group, pull, put, sort, handle, filter, component, forceFallback) {
    var sortable = new Sortable(document.getElementById(id), {
        animation: 200,
        group: {
            name: group,
            pull: pull || true,
            put: put
        },
        filter: filter || undefined,
        sort: sort,
        forceFallback: forceFallback,
        handle: handle || undefined,
        onUpdate: (event) => {
            event.item.remove();
            event.to.insertBefore(event.item, event.to.childNodes[event.oldIndex]);

            component.invokeMethodAsync('OnUpdateJS', event.oldDraggableIndex, event.newDraggableIndex);
        },
        onRemove: (event) => {
            if (event.pullMode === 'clone') {
                event.clone.remove();
            }

            event.item.remove();
            event.from.insertBefore(event.item, event.from.childNodes[event.oldIndex]);

            component.invokeMethodAsync('OnRemoveJS', event.oldDraggableIndex, event.newDraggableIndex);
        },
        onAdd: function (/**Event*/evt) {
            var itemEl = evt.item;  // dragged HTMLElement
            evt.to;    // target list
            evt.from;  // previous list
            evt.oldIndex;  // element's old index within old parent
            evt.newIndex;  // element's new index within new parent
            evt.oldDraggableIndex; // element's old index within old parent, only counting draggable elements
            evt.newDraggableIndex; // element's new index within new parent, only counting draggable elements
            evt.clone // the clone element
            evt.pullMode;  // when item is in another sortable: `"clone"` if cloning, `true` if moving
            component.invokeMethodAsync('OnAddJS', evt.oldDraggableIndex, evt.newDraggableIndex, evt.from.id, evt.to.id,);
        }
    });
}

File: CalendarFluent.UI/Components/SortableList.razor.cs
using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;
using System.Diagnostics.CodeAnalysis;
using static CalendarFluent.UI.Components.Pages.Home;

namespace CalendarFluent.UI.Components
{
	public partial class SortableList<T> where T : Item
	{

		[Parameter]
		public RenderFragment<T>? SortableItemTemplate { get; set; }

		[Parameter, AllowNull]
		public List<T> Items { get; set; }

		[Parameter]
		public EventCallback<(int oldIndex, int newIndex)> OnUpdate { get; set; }

		[Parameter]
		public EventCallback<(int oldIndex, int newIndex)> OnRemove { get; set; }

		[Parameter]
		public EventCallback<(int oldIndex, int newIndex, string from, string to)> OnAdd { get; set; }

		[Parameter]
		public string Id { get; set; } = Guid.NewGuid().ToString();

		[Parameter]
		public string Group { get; set; } = Guid.NewGuid().ToString();

		[Parameter]
		public string? Pull { get; set; }

		[Parameter]
		public bool Put { get; set; } = true;

		[Parameter]
		public bool Sort { get; set; } = true;

		[Parameter]
		public string Handle { get; set; } = string.Empty;

		[Parameter]
		public string? Filter { get; set; }

		[Parameter]
		public bool ForceFallback { get; set; } = true;

		private DotNetObjectReference<SortableList<T>>? _selfReference;

		protected override async Task OnAfterRenderAsync(bool firstRender)
		{
			if (firstRender)
			{
				_selfReference = DotNetObjectReference.Create(this);
				var module = await JS.InvokeAsync<IJSObjectReference>("import", "./Components/SortableList.razor.js");
				await module.InvokeAsync<string>("init", Id, Group, Pull, Put, Sort, Handle, Filter, _selfReference, ForceFallback);
			}
		}

		[JSInvokable]
		public void OnUpdateJS(int oldIndex, int newIndex)
		{
			OnUpdate.InvokeAsync((oldIndex, newIndex));
		}

		[JSInvokable]
		public void OnRemoveJS(int oldIndex, int newIndex)
		{
			OnRemove.InvokeAsync((oldIndex, newIndex));
		}
		[JSInvokable]
		public void OnAddJS(int oldIndex, int newIndex, string from, string to)
		{
			OnAdd.InvokeAsync((oldIndex, newIndex, from, to));
		}
		public void Dispose() => _selfReference?.Dispose();
	}
}